lib/core/buckets/usecase.ts

```diff
@@ -213,6 +244,157 @@ export class BucketsUsecase {
     return (usage && usage.total) || 0;
   } 
 
+  async startUpload(
+    userId: string,
+    bucketId: string,
+    cluster: string[],
+    uploads: { index: number; size: number }[],
+    auth: { username: string; password: string },
+    multiparts: number = 1
+  ) {
+    const [bucket, user] = await Promise.all([
+      this.bucketsRepository.findOne({ id: bucketId }),
+      this.usersRepository.findById(userId),
+    ]);
+
+    if (!user) {
+      throw new UserNotFoundError();
+    }
+
+    if (!bucket) {
+      throw new BucketNotFoundError();
+    }
+
+    if (bucket.user !== userId) {
+      throw new BucketForbiddenError();
+    }
+
+    if (!uploads) {
```

Comment:
Maybe this should be checked on the controller, as the uploads are expected to be received here (do not have the ? in the type, and should not, as the use case is not responsible for validating the input)

Reply:
yep, fixed